sub-packages/bionemo-amplify/src/bionemo/amplify/dataset.py

```diff
@@ -0,0 +1,147 @@
+                that __getitem__ is deterministic, but can be random across different runs. If None, a random seed is
+                generated.
+            max_seq_length: Crop long sequences to a maximum of this length, including BOS and EOS tokens.
+            mask_prob: The overall probability a token is included in the loss function. Defaults to 0.15.
+            mask_token_prob: Proportion of masked tokens that get assigned the <MASK> id. Defaults to 0.8.
+            mask_random_prob: Proportion of tokens that get assigned a random natural amino acid. Defaults to 0.1.
+            random_mask_strategy: Whether to replace random masked tokens with all tokens or amino acids only. Defaults to RandomMaskStrategy.AMINO_ACIDS_ONLY.
+            tokenizer: The input AMPLIFY tokenizer. Defaults to the standard AMPLIFY tokenizer.
+        """
+        self.protein_dataset = hf_dataset
+        self.total_samples = len(self.protein_dataset)
+        self.seed = seed
+        self.max_seq_length = max_seq_length
+        self.random_mask_strategy = random_mask_strategy
+        if tokenizer is None:
+            self.tokenizer = BioNeMoAMPLIFYTokenizer()
+        else:
+            self.tokenizer = tokenizer
+
+        if self.tokenizer.mask_token_id is None:
+            raise ValueError("Tokenizer does not have a mask token.")
+
+        self.mask_config = masking.BertMaskConfig(
+            tokenizer=self.tokenizer,
+            random_tokens=range(self.tokenizer.vocab_size)
+            if self.random_mask_strategy == RandomMaskStrategy.ALL_TOKENS
+            else range(6, self.tokenizer.vocab_size),
+            mask_prob=mask_prob,
+            mask_token_prob=mask_token_prob,
+            random_token_prob=mask_random_prob,
+        )
+
+    def __len__(self) -> int:
+        """Returns the total number of sequences in the dataset."""
+        return self.total_samples
+
+    def __getitem__(self, index: EpochIndex) -> BertSample:
+        """Deterministically masks and returns a protein sequence from the dataset.
+
+        Args:
+            index: The current epoch and the index of the cluster to sample.
+
+        Returns:
+            A (possibly-truncated), masked protein sequence with CLS and EOS tokens and associated mask fields.
+        """
+        # Initialize a random number generator with a seed that is a combination of the dataset seed, epoch, and index.
+        rng = np.random.default_rng([self.seed, index.epoch, index.idx])
+        if index.idx >= len(self):
+            raise IndexError(f"Index {index.idx} out of range [0, {len(self)}).")
+
+        sequence = self.protein_dataset[int(index.idx)]["sequence"]
+
+        # We don't want special tokens before we pass the input to the masking function; we add these in the collate_fn.
+        tokenized_sequence = self._tokenize(sequence)
+        cropped_sequence = _random_crop(tokenized_sequence, self.max_seq_length, rng)
```

Comment:
This could use a comment.

Reply:
added, but this was largely copied from the ESM-2 dataset implementation.